subprojects/core-api/src/main/java/org/gradle/api/tasks/util/internal/PatternSpecFactory.java

```diff
@@ -108,4 +128,25 @@
 
         return new RelativePathSpec(matcher);
     }
+
+    private enum CaseSensitivity {
+        CASE_SENSITIVE(true),
+        CASE_INSENSITIVE(false);
+
+        CaseSensitivity(boolean caseSensitive) {
+            this.caseSensitive = caseSensitive;
+        }
+
+        public static CaseSensitivity forCaseSensitive(boolean caseSensitive) {
+            return caseSensitive
+                ? CASE_SENSITIVE
+                : CASE_INSENSITIVE;
+        }
+
+        private final boolean caseSensitive;
+
+        public boolean isCaseSensitive() {
+            return caseSensitive;
```

Comment:
I like the field better, since I think the methods in one enum constant shouldn't reference the other enum constants. I could also do the implementation of `isCaseSensitive` in the enum constants themselves. WDYT?

Reply:
> methods in one enum constant shouldn't reference the other enum constants

Where does this come from? Just curious, I'm not feeling strongly about this.

Reply:
>> methods in one enum constant shouldn't reference the other enum constants
>
> Where does this come from? Just curious, I'm not feeling strongly about this.

No particular source, it just feels strange to me to reference to anything outside the current instance if I can avoid it.